simulator/modules/decode/decode.cpp

```diff
@@ -87,6 +96,24 @@ void Decode<FuncInstr>::clock( Cycle cycle)
 
     auto instr = read_instr( cycle);
 
+    /* acquiring real information for BPU */
+    wp_bp_update->write( instr.get_bp_upd(), cycle);
+
+    bool is_misprediction = instr.is_direct_jump() &&
```

Comment:
What would be interesting here is to add a command line option to switch between old and new behavior. It would allow to demonstrate performance changes easily.